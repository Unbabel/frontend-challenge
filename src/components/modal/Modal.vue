<script>
	// Store
	import { mapState, mapActions } from "vuex";

	export default {
		methods: {
			...mapActions({
				toggleModalVisibility: "toggleModalVisibility",
			}),
			closeModal() {
				this.toggleModalVisibility();
			},
		},
		computed: {
			...mapState({
				showModal: "showModal",
			}),
		},
	};
</script>

<template>
	<div id="custom-modal" class="modal" :class="showModal && 'modal-active'">
		<div class="modal-content">
			<div class="flex-row-end">
				<span class="close" @click="closeModal">&times;</span>
			</div>
			<div class="modal-slot-content">
				<slot></slot>
			</div>
		</div>
	</div>
</template>

<style scoped>
	.modal {
		display: none; /* Hidden by default */
		position: fixed;
		z-index: 11;
		left: 0;
		top: 0;
		width: 100%;
		height: 100%;
		overflow: auto;
		background-color: var(--palette-color); /* Fallback color */
		background-color: rgba(0, 0, 0, 0.4); /* Black w/ opacity */
	}

	.modal-active {
		display: block;
	}

	.modal-content {
		background-color: #fefefe;
		margin: 15% auto;
		padding: 1.25rem;
		border: 1px solid #888;
		width: fit-content;
		min-width: 25%;
	}

	.modal-slot-content {
		margin: 0rem 2rem 0.5rem 2rem;
	}

	.flex-row-end {
		display: flex;
		flex-direction: row;
		justify-content: end;
	}

	.close {
		color: #aaa;
		float: right;
		font-size: 1.5rem;
		font-weight: bold;
	}

	.close:hover,
	.close:focus {
		color: black;
		text-decoration: none;
		cursor: pointer;
	}
</style>
