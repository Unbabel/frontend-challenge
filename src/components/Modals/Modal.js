import * as React from 'react'
import '../../resources/styles/modal.scss'

const Modal = ({ onClose }) => {
    return (<React.Fragment>
        <div className="modal">
                <div className="modal-content">
                    <span onClick={onClose} className="close">&times;</span>
                    <p>Data Is Uploaded..</p>
                </div>
        </div>
    </React.Fragment>
    )
}
export default Modal;