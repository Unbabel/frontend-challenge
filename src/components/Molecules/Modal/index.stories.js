import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import Modal from '.'

storiesOf('Molecules/Modal', module)
  .add('default', () => (
    <Modal onClose={action('closed')} isOpen>
"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua    </Modal>
  ))
  .add('with title', () => (
    <Modal onClose={action('closed')} title="Hello" isOpen>
"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua    </Modal>
  ))
  .add('closeable', () => (
    <Modal onClose={action('closed')} title="Hello" closeable isOpen>
"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua    </Modal>
  ))
