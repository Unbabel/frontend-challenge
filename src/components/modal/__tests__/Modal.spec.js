import Vuex from "vuex";
import { describe, it, expect, beforeEach, vi } from "vitest";
import { mount, createLocalVue } from "@vue/test-utils";

// Components
import Modal from "../Modal.vue";

const localVue = createLocalVue();
localVue.use(Vuex);

const store = new Vuex.Store({
	state: {
		showModal: true,
	},
});

describe("Modal", () => {
	let wrapper;

	beforeEach(() => {
		// Arrange
		wrapper = mount(Modal, { localVue, store });
	});

	it("matches snapshot", () => {
		// Assert
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("renders properly", () => {
		// Act
		const divs = wrapper.findAll("div");
		const spans = wrapper.findAll("span");

		// Assert
		expect(divs.exists()).toBe(true);
		expect(divs).toHaveLength(4);
		expect(spans.exists()).toBe(true);
		expect(spans).toHaveLength(1);
	});

	[
		{ element: "div", clx: "modal" },
		{ element: "div", clx: "modal-active" },
		{ element: "div", clx: "modal-content" },
		{ element: "div", clx: "flex-row-end" },
		{ element: "span", clx: "close" },
		{ element: "div", clx: "modal-slot-content" },
	].forEach(({ element, clx }) =>
		it("has correct css classes", () => {
			// Act
			const elements = wrapper.findAll(element);
			const filteredElements = elements.filter((el) => el.classes(clx));

			// Assert
			expect(filteredElements).toHaveLength(1);
		})
	);

	it("should call closeModal on click", async () => {
		// Arrange
		const spy = vi.spyOn(Modal.methods, "closeModal");
		wrapper = mount(Modal, { localVue, store });

		// Act
		const span = await wrapper.find("span");
		await span.trigger("click");

		// Assert
		expect(spy).toBeCalled();
	});
});
