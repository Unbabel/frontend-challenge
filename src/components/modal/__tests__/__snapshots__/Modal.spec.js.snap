// Vitest Snapshot v1

exports[`Modal > matches snapshot 1`] = `
"<div data-v-b759700c=\\"\\" id=\\"custom-modal\\" class=\\"modal modal-active\\">
  <div data-v-b759700c=\\"\\" class=\\"modal-content\\">
    <div data-v-b759700c=\\"\\" class=\\"flex-row-end\\"><span data-v-b759700c=\\"\\" class=\\"close\\">Ã—</span></div>
    <div data-v-b759700c=\\"\\" class=\\"modal-slot-content\\"></div>
  </div>
</div>"
`;
